allennlp/models/reading_comprehension/naqanet.py

```diff
@@ -0,0 +1,498 @@
+            self._question_span_start_predictor = FeedForward(modeling_out_dim * 2,
+                                                              activations=[Activation.by_name('relu')(),
+                                                                           Activation.by_name('linear')()],
+                                                              hidden_dims=[modeling_out_dim, 1],
+                                                              num_layers=2)
+            self._question_span_end_predictor = FeedForward(modeling_out_dim * 2,
+                                                            activations=[Activation.by_name('relu')(),
+                                                                         Activation.by_name('linear')()],
+                                                            hidden_dims=[modeling_out_dim, 1],
+                                                            num_layers=2)
+
+        if "addition_subtraction" in answering_abilities:
+            self._addition_subtraction_index = self.answering_abilities.index("addition_subtraction")
+            self._number_sign_predictor = FeedForward(modeling_out_dim * 3,
+                                                      activations=[Activation.by_name('relu')(),
+                                                                   Activation.by_name('linear')()],
+                                                      hidden_dims=[modeling_out_dim, 3],
+                                                      num_layers=2)
+
+        if "counting" in answering_abilities:
+            self._counting_index = self.answering_abilities.index("counting")
+            self._count_number_predictor = FeedForward(modeling_out_dim,
+                                                       activations=[Activation.by_name('relu')(),
+                                                                    Activation.by_name('linear')()],
+                                                       hidden_dims=[modeling_out_dim, 10],
+                                                       num_layers=2)
+
+        self._drop_metrics = DropEmAndF1()
+        self._dropout = torch.nn.Dropout(p=dropout_prob)
+
+        initializer(self)
+
+    def forward(self,  # type: ignore
+                question: Dict[str, torch.LongTensor],
+                passage: Dict[str, torch.LongTensor],
+                numbers_in_passage: Dict[str, torch.LongTensor],
+                number_indices: torch.LongTensor,
+                answer_as_passage_spans: torch.LongTensor = None,
+                answer_as_question_spans: torch.LongTensor = None,
+                answer_as_add_sub_expressions: torch.LongTensor = None,
+                answer_as_counts: torch.LongTensor = None,
+                metadata: List[Dict[str, Any]] = None) -> Dict[str, torch.Tensor]:
+        # pylint: disable=arguments-differ, unused-argument
```

Comment:
May be add a docstring with short description of what's happening here and what the parameters mean? Particularly, I do not understand why `numbers_in_passage` is needed since it's not being used.

Reply:
I put a better note in the class docstring, and I also wondered why we need `numbers_in_passage` and looked into it.  I'll add a comment there too.

Reply:
Actually, I was afraid of modifying it because I thought it would break the dataset reader, but I just did, and nothing broke.  So I removed the unused field.